PHPC-2593: Update libmongoc to 1.30.5

alcaeus opened the pull request:
PHPC-2593

alcaeus committed:
PHPC-2593: Update libmongoc to 1.30.5

alcaeus committed:
Update tests for new BSON validation logic

GromNaN reviewed (changes requested):
According to the errors in evergreen, some exception messages have changed.
https://github.com/mongodb/mongo-c-driver/compare/80fa63e3d11454c21e93b6bf8d262725011ff575...252989c310f7a326230f66311630ba259e34e4b3#diff-8ae42945286d636565d9a98bbf03a7528fb0fcdeddfb3809ee76899fd2032ff0R1973

alcaeus commented:
Yep. I confirmed with the C team whether the missing validation for invalid keys is expected, and they confirmed it is. I updated the tests to test for the new exception message and removed tests asserting invalid key behaviour.
